src/front/wgsl/mod.rs

```diff
@@ -1204,6 +1215,8 @@ impl std::error::Error for ParseError {
 
 pub struct Parser {
     scopes: Vec<(Scope, usize)>,
+    module_scope_variable_identifiers: FastHashMap<String, Span>,
```

Comment:
do these need to be separate from function identifiers?

Reply:
Maybe I misread the spec, but I can't find confirmation that module scope functions and variables cannot have the same name. If they can't have the same identifier, we can merge 2 hashmaps.

Reply:
Yes, pretty sure they are in the same namespace. Let's merge the maps.